docs/architecture/high-level-components.md

```diff
@@ -0,0 +1,27 @@
+# Overview
+
+This document describes communication between Management Plane, Application and Runtimes. 
+
+## Components
+Management Plane is a central place which stores configuration about Applications and Runtimes and is responsible for propagating that information to Applications and Runtimes. 
+Management Plane plays crucial role in establishing trusted connection between Applications and Runtimes. 
+
+![](./assets/components-high-level.svg)
+
+An administrator adds Runtimes, Applications and configures them through Management Plane GraphQL API. 
+To keep Runtimes in-sync with the configuration, an agent installed on a Runtime is responsible for fetching actual configuration. 
+An Agent sends information about Runtime healthchecks to Management Plane. 
+In case an Application has configured optional Webhooks, Management Plane notify an Application about any events that concern given Application.
```

Comment:
```suggestion
In case an Application has configured optional Webhooks, Management Plane notifies an Application about any events that concern given Application.
```